engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/bool/ToBoolNode.java

```diff
@@ -0,0 +1,37 @@
+package org.enso.interpreter.node.expression.builtin.bool;
+
+import com.oracle.truffle.api.dsl.Fallback;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.nodes.Node;
+import com.oracle.truffle.api.nodes.NodeInfo;
+import org.enso.interpreter.dsl.MonadicState;
+import org.enso.interpreter.runtime.Context;
+import org.enso.interpreter.runtime.error.DataflowError;
+import org.enso.interpreter.runtime.error.PanicException;
+import org.enso.interpreter.runtime.state.Stateful;
+
+@NodeInfo(description = "Helper node to handle Bool values that can also return DataflowErrors")
+public abstract class ToBoolNode extends Node {
+
+  public static ToBoolNode build() {
+    return ToBoolNodeGen.create();
+  }
+
+  abstract Stateful execute(@MonadicState Object state, Object value);
```

Comment:
This should not return a `Stateful`, it's not the node's responsibility – the point of this node is "make sure my arg is a bool", so just return a boolean.